### PR TITLE
Better format and extension handling

### PR DESCRIPTION
**Pull request by mikkoh**
Currently in the release version there's a bug where if you don't set a image format and pass an output path with `.jpg` it will complain that the extension and format don't match.

This PR makes format and extension handling more friendly.

The rules are as follows:

1. Infer the format/mime-type from the extension and ignore the format if passed
2. If a format cannot be inferred then use the passed in format (default png)
3. If a format cannot be inferred and the passed in format is incorrect then throw an exception

### CODE REVIEW

File: src/parse-output-path-and-format.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export function parseOutputPathAndFormat(output: string, format: string) {
+  const extension = path.extname(output);
+  const extensionMimeType = mime.lookup(extension);
+
+  if (extensionMimeType === format) {
+    return [output, format];
+  }
+
+  const formatExtension = mime.extension(format);
```
Comment (tylerrowsell) [inline src/parse-output-path-and-format.ts:12]: I wonder if it is worth also adding the ability to infer format from extension. i.e. `.jpg` and we know it needs to be `image/jpeg`

**Review by tylerrowsell — APPROVED** [review]
Just one potential improvement, but I could go either way on it.

File: src/parse-output-path-and-format.ts (hunk 1 above)
Comment (mikkoh) [reply src/parse-output-path-and-format.ts:12]: Yeah this happens now. Although it's not clear from the code. Let me actually make it a bit cleaner...

Currently we have this:
```typescript
  if (!extensionMimeType) {
    const extension = formatExtension === "jpeg" ? "jpg" : formatExtension;

    return [`${output}.${extension}`, format];
  }

  return [output, extensionMimeType];
```

I'll remove the negation and check that the extension produces a valid format.

**Comment by mikkoh** [comment]
Gah. I don't think Shipit liked that I did the version bump and adding the feature in the same PR.

I'm gonna revert and try again.

**Pull request merged**